expression/builtin_string.go

```diff
@@ -2959,18 +2959,42 @@ func (c *fromBase64FunctionClass) getFunction(ctx sessionctx.Context, args []Exp
 	}
 	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETString)
 	bf.tp.Flen = mysql.MaxBlobWidth
+
+	valStr, _ := ctx.GetSessionVars().GetSystemVar(variable.MaxAllowedPacket)
+	maxAllowedPacket, err := strconv.ParseUint(valStr, 10, 64)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
 	types.SetBinChsClnFlag(bf.tp)
-	sig := &builtinFromBase64Sig{bf}
+	sig := &builtinFromBase64Sig{bf, maxAllowedPacket}
 	return sig, nil
 }
 
+// base64NeededDecodedLength return the base64 decoded string length.
+func base64NeededDecodedLength(n int) int {
+	// Returns -1 indicate the result will overflow.
+	if strconv.IntSize == 64 {
+		if n > 3074457345618258602 {
+			return -1
+		}
+	} else {
+		if n > 715827882 {
+			return -1
+		}
+	}
+	return n * 3 / 4
```

Comment:
How about `n/4*3`, then the above validation can be removed?

Reply:
when `n%4 > 4/3` , `n/4*3` is not equal to `n*3/4`.
mysql calculates decode length by `n*3/4`, I think we'd better use the same formula.

Reply:
ok